### PR TITLE
refactor(add user form): remove validation and error msgs

### PR DESCRIPTION
**Pull request by domsteinbach**
resolves DEV-3337

### CODE REVIEW

**Review by irmastnt — CHANGES_REQUESTED** [review]
We are adding language translations to all new code from the last week.

**Comment by domsteinbach** [comment]
> We are adding language translations to all new code from the last week.

Good to know, but what does that imply? This code is from now and not last week. Is there any inconvenience for adding/changing translations? What are the changes requested?

**Comment by domsteinbach** [comment]
The translations are introduced.

File: apps/dsp-app/src/app/project/collaboration/add-user/add-user.component.html [hunk 1]
```diff
@@ -1,26 +1,22 @@
...
+    <h2 class="mat-headline-6">{{ 'appLabels.addProjectMember.title' | translate }}</h2>
 
     <span class="fill-remaining-space"></span>
 
     <span class="app-toolbar-action select-form">
-      <form [formGroup]="selectUserForm" class="form" *ngIf="!loading">
+      <form [formGroup]="selectUserForm" class="form">
         <!-- auto complete list to select user -->
         <div class="form-content">
           <mat-form-field class="large-field select-user">
```
Comment (irmastnt) [inline apps/dsp-app/src/app/project/collaboration/add-user/add-user.component.html:14]: mat-label is missing

**Review by irmastnt — APPROVED** [review]

File: apps/dsp-app/src/app/project/collaboration/add-user/add-user.component.ts [hunk 2]
```diff
@@ -1,210 +1,115 @@
...
-  /**
-   * form error hints
-   */
-  validationMessages = {
-    addUser: {
-      existingName: "This user is already a member of the project. You can't add him / her.",
-    },
-  };
+  private readonly _destroyed$: Subject<void> = new Subject<void>();
+  selectUserForm: FormGroup;
```
[the remark this answers is not in the document]
Comment (domsteinbach) [reply apps/dsp-app/src/app/project/collaboration/add-user/add-user.component.ts:24]: Here not. ts allowes class props to be declared without initialisation as long they are initialised before any usage. As ngOnInit is guaranteed to run before any component methods that rely on selectUserForm, it is safe to initialize the form there and there is no complain.

Here it is better without an assertion as if there are any changes in the business logic an error is thrown if it breaks. If we use "?" or "!" there will be no complain but maybe a bug.
In general we should not use "?" or "!" if there is really no other way. In most of the cases where an assertion is used, the component is not made nicely imho.

File: apps/dsp-app/src/app/project/collaboration/add-user/add-user.component.ts [hunk 3]
```diff
@@ -1,210 +1,115 @@
...
-
-  buildForm(): void {
-    // clean autocomplete list
-    this.users = [];
-
-    // get all users
-    this._dspApiConnection.admin.usersEndpoint.getUsers().subscribe((response: ApiResponseData<UsersResponse>) => {
-      this.setSelectUserForm(response.body.users);
+    this.selectUserForm = this.fb.group({
+      addUser: new FormControl<ReadUser | null>(null),
```
[the remark this answers is not in the document]
Comment (domsteinbach) [reply apps/dsp-app/src/app/project/collaboration/add-user/add-user.component.ts:53]: I prefer the more verbose (for me) here :-)

**Pull request merged**